### PR TITLE
benetech/ServiceNet#943

### PR DESCRIPTION
**Pull request by kdondziak**

### CODE REVIEW

File: src/main/webapp/app/app-with-provider.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import React from 'react';
```
Comment (kdondziak) [inline src/main/webapp/app/app-with-provider.tsx]: Name for this file may be better. tell me if you have a better idea.

Comment (awalkowiak) [reply src/main/webapp/app/app-with-provider.tsx]: Why this is in a separate component? Is it really required to have a separate app component?

File: src/main/webapp/app/modules/provider-site/avatar.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import './provider-site.scss';
+import React from 'react';
+
+export interface IAvatar {
+  name: string;
+  size: 'big' | 'small';
+}
+
+export const Avatar: React.FC<IAvatar> = props => <div className={`avatar-${props.size}`}>{props.name}</div>;
```
Comment (awalkowiak) [inline src/main/webapp/app/modules/provider-site/avatar.tsx]: Do you need the interface? Out stateless components don't have interfaces.

**Review by awalkowiak — CHANGES_REQUESTED** [review]
- The hamburger menu is broken on a responsive view,
- Hamburger menu should include all menu items listed directly in it without any additional dropdown,
- There should be no Admin menu on the header when the hamburger menu is active (imho it should not be accessible from a header in responsive mode at all, it should be accessible from hamburger menu directly)


From the other notes:
Carousel looks good on both desktop and responsive view, good job

File: src/main/webapp/app/modules/provider-site/provider-site.tsx [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    this.setState({ menuOpen: !this.state.menuOpen });
+  };
+
+  render() {
+    const { userLogin } = this.props;
+    const { menuOpen } = this.state;
+
+    const HeaderComponent = (
+      <div>
+        <MediaQuery maxDeviceWidth={768}>
```
Comment (awalkowiak) [inline src/main/webapp/app/modules/provider-site/provider-site.tsx]: What will happen if the width will be 768? maxDeviceWidth and minDeviceWidth are the same. Also what happens in a horizontal view?

File: src/main/webapp/app/modules/provider-site/user-records.tsx [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+          <FontAwesomeIcon size="lg" icon="chevron-right" />
+        </div>
+      </div>
+    </div>
+  );
+
+  render() {
+    const record = {
+      organization: {
+        id: '4e63866c-a6a3-4010-92bb-18b4cd52a1c3',
```
Comment (awalkowiak) [inline src/main/webapp/app/modules/provider-site/user-records.tsx]: I think this data should be randomized and should live in a separate file.

File: src/main/webapp/app/app.tsx [hunk 5]
```diff
@@ -38,8 +39,8 @@ export class App extends React.Component<IAppProps> {
 
   render() {
     const padding = '10px';
-    return (
-      <Router>
+    const oldApp = (
```
Comment (awalkowiak) [inline src/main/webapp/app/app.tsx]: I am not sure about the naming "oldApp" it still gonna be a part of the app.

**Review by awalkowiak — CHANGES_REQUESTED** [review]

File: src/main/webapp/app/modules/provider-site/provider-site.tsx (hunk 3 above)
Comment (kdondziak) [reply src/main/webapp/app/modules/provider-site/provider-site.tsx]: You are right, I'll change this

File: src/main/webapp/app/app.tsx [hunk 6]
```diff
@@ -15,12 +15,13 @@ import { getProfile } from 'app/shared/reducers/application-profile';
...
 import Header from 'app/shared/layout/header/header';
 import Footer from 'app/shared/layout/footer/footer';
-import { hasAnyAuthority } from 'app/shared/auth/private-route';
+import PrivateRoute, { hasAnyAuthority } from 'app/shared/auth/private-route';
 import ErrorBoundary from 'app/shared/error/error-boundary';
 import { AUTHORITIES } from 'app/config/constants';
 import AppRoutes from 'app/routes';
 import GoBackButton from 'app/shared/layout/go-back-button';
 import { containerStyle } from 'app/shared/util/measure-widths';
+import ProviderSite from 'app/modules/provider-site/provider-site';
```
Comment (jkondrat) [inline src/main/webapp/app/app.tsx]: couldn't we call it ProviderHome or ProviderLandingPage? and move all the related modules to 'provider' package

File: src/main/webapp/app/app.tsx [hunk 7]
```diff
@@ -72,6 +73,20 @@ export class App extends React.Component<IAppProps> {
           </div>
           <Footer />
         </div>
+      </div>
+    );
+
+    return (
+      <Router>
+        <Switch>
+          <PrivateRoute
```
Comment (jkondrat) [inline src/main/webapp/app/app.tsx:82]: the measure-layer utility might break for the new route if we won't have measure-layer div there

File: src/main/webapp/app/modules/provider-site/avatar.tsx [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+import React from 'react';
+
+export interface IAvatar {
+  name: string;
+  size: 'big' | 'small';
+  mobile?: boolean;
+}
+
+export const Avatar: React.FC<IAvatar> = props => (
+  <div className={`avatar-${props.mobile ? 'mobile-' : ''}${props.size}`}>{props.name}</div>
```
Comment (jkondrat) [inline src/main/webapp/app/modules/provider-site/avatar.tsx]: perhaps we could have mobile class at some top-level component?

File: src/main/webapp/app/modules/home/record-card.tsx [hunk 9]
```diff
@@ -51,9 +51,6 @@ class RecordCard extends React.Component<IRecordCardProps, IRecordCardState> {
   };
 
   componentDidMount() {
-    if (!this.props.loadingUser && !this.props.user.id) {
```
Comment (jkondrat) [inline src/main/webapp/app/modules/home/record-card.tsx:54]: are you sure we don't need to fetch the user?

**Review by jkondrat — CHANGES_REQUESTED** [review]

**Pull request merged**